app/code/Magento/Inventory/Setup/Operation/CreateReservationTable.php

```diff
@@ -0,0 +1,127 @@
+
+        $setup->getConnection()->createTable($reservationTable);
+    }
+
+    /**
+     * @param SchemaSetupInterface $setup
+     * @return Table
+     */
+    private function createReservationTable(SchemaSetupInterface $setup)
+    {
+        $reservationTable = $setup->getTable(ReservationResourceModel::TABLE_NAME_RESERVATION);
+        $stockTable = $setup->getTable(StockResourceModel::TABLE_NAME_STOCK);
+
+        return $setup->getConnection()->newTable(
+            $reservationTable
+        )->setComment(
+            'Inventory Reservation Table'
+        )->addColumn(
+            ReservationInterface::RESERVATION_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_IDENTITY => true,
+                Table::OPTION_UNSIGNED => true,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_PRIMARY => true,
+            ],
+            'Reservation ID'
+        )->addColumn(
+            ReservationInterface::STOCK_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_UNSIGNED => true,
+            ],
+            'Stock ID'
+        )->addColumn(
+            ReservationInterface::SKU,
+            Table::TYPE_TEXT,
+            64,
+            [
+                Table::OPTION_NULLABLE => false,
+            ],
+            'Sku'
+        )->addColumn(
+            ReservationInterface::QUANTITY,
+            Table::TYPE_DECIMAL,
+            null,
+            [
+                Table::OPTION_UNSIGNED => false,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_DEFAULT => 0,
+            ],
+            'Quantity'
+        )->addColumn(
+            ReservationInterface::STATUS,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_UNSIGNED => true,
+            ],
+            'Status'
+        )->addForeignKey(
+            $setup->getFkName(
+                $reservationTable,
+                ReservationInterface::STOCK_ID,
+                $stockTable,
+                StockInterface::STOCK_ID
+            ),
+            ReservationInterface::STOCK_ID,
+            $stockTable,
+            StockInterface::STOCK_ID,
+            AdapterInterface::FK_ACTION_CASCADE
+        )->addForeignKey(
+            $setup->getFkName(
+                $reservationTable,
+                ReservationInterface::SKU,
+                'catalog_product_entity',
+                'sku'
+            ),
+            ReservationInterface::SKU,
+            'catalog_product_entity',
+            'sku',
+            AdapterInterface::FK_ACTION_CASCADE
+        )->addIndex(
+            $setup->getIdxName(
+                $reservationTable,
+                [
+                    ReservationInterface::STOCK_ID,
+                    ReservationInterface::SKU,
```

Comment:
I propose to extend this index and add Qty as well.

because executing such query 
```sql
select 
   MAX(r.status) as grouped_status, group_concat(reservation_id) as reservations
from 
   Reservations as r
group by
   stock_id, sku, qty
having
   grouped_status > 100
```
MySQL could use such index for Groupping operation , which is made by (stock_id, sku, qty)

Also this index would be used for Search lookup by  (stock_id, sku)